visualisation/hotspotMaps/server.R

```diff
@@ -102,39 +92,12 @@ shinyServer(function(input, output, session) {
 
   }, height = 500)
   
-  # output$speciesOccurrenceMap <- renderPlot({
-  #   
-  #   if (input$selectAbsences == TRUE) {
-  #     dataToPlot <- processedDataCompiled[processedDataCompiled$simpleScientificName == input$speciesOccurrence,] 
-  #   } else {
-  #     dataToPlot <- processedDataCompiled[processedDataCompiled$simpleScientificName == input$speciesOccurrence &
-  #                                           processedDataCompiled$individualCount == 1,]
-  #   }
-  #   
-  #   basePlot <- ggplot(regionGeometry) +
-  #     geom_sf(fill = "white", lwd = 0.7) +
-  #     theme_classic() +
-  #     theme(legend.text=element_text(size=14),
-  #           legend.position = "bottom",
-  #           legend.direction = "vertical")
-  #   
-  #   if (input$dataClassification == "dataSource") {
-  #     fullPlot <- basePlot + geom_sf(data = dataToPlot, aes(colour = colours1, labels = name)) +
-  #       scale_colour_identity(guide = "legend", breaks = colourFrame1$colour, 
-  #                             labels = str_wrap(colourFrame1$name, width = 50))
-  #   } else {
-  #     fullPlot <- basePlot + geom_sf(data = dataToPlot, aes(colour = colours2, labels = dataType)) +
-  #       scale_colour_identity(guide = "legend", breaks = colourFrame2$colour, labels = colourFrame2$dataTypeFull, name = "Data source")
-  #   }
-  #   
-  #   fullPlot
-  #   
-  # }, height = 600)
-  
   output$covariateMap <- renderPlot({
 
-    covariateToPlot <- covariateData[[input$covariate]]
-    covariateDataDF <- as.data.frame(covariateToPlot, xy = TRUE)
+    covariateToPlot <- terra::crop(covariateData[[input$covariate]],
+                                   project(vect(regionGeometry), covariateData),
+                                   mask = T)
+    covariateDataDF <- as.data.frame(project(covariateToPlot, "+proj=longlat +datum=WGS84 +no_defs +ellps=WGS84 +towgs84=0,0,0"), xy = TRUE)
```

Comment:
I think projection of all things being plotted should be the working projection specified in masterScript.R; if this is reapplied to other regions, transforming to lat/long may not be appropriate.